src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -2183,7 +2298,7 @@ void AutomationInstrumentClipView::modEncoderAction(int32_t whichModEncoder, int
 	//if user holding a node down, we'll adjust the value of the selected parameter being automated
 	if (isUIModeActive(UI_MODE_NOTES_PRESSED) || padSelectionOn) {
 
-		if (clip->lastSelectedParamID != kNoLastSelectedParamID
```

Comment:
I think this function would benefit from splitting into a few smaller functions for each state instead of the nested if statemenst